### PR TITLE
CMake format

### PR DESCRIPTION
**Pull request by renemilk**
I've stumbled across a tool to end our cmake formatting woes: https://github.com/cheshirekow/cmake_format/
I've already integrated it into our cmake setup and all dependent module will automagically inherit the mechanism. Only thing needed would be a .vcsetup bump after we've moved the tool's config there. 
There's however a, hopefully small, issue atm with one of our files: https://github.com/cheshirekow/cmake_format/issues/28
Other than that the most work left would be finding the right options for us :wink:

### CODE REVIEW

**Comment by renemilk** [comment]
The application of the format now works, thanks to https://github.com/cheshirekow/cmake_format/issues/28 being essentially fixed. 

However this currently breaks license header comments. One way to prevent this would be for us to envelop the header like this, making cmake_format reproduce it verbatim:
```
# ~~~
# the license stuff
# ~~~
```
Assuming this is doable with our pylicense code, I'm ok with that.

Other than that there's currently no fixing of our randomized upper-lower casing for function names and commands, which I would really like.

**Comment by ftalbrecht** [comment]
On March 29, 2018 1:38:11 PM GMT+02:00, "René Milk" <notifications@github.com> wrote:
>The application of the format now works, thanks to
>https://github.com/cheshirekow/cmake_format/issues/28 being essentially
>fixed. 
>
>However this currently breaks license header comments. One way to
>prevent this would be for us to envelop the header like this, making
>cmake_format reproduce it verbatim:
>```
># ~~~
># the license stuff
># ~~~
>```
>Assuming this is doable with our pylicense code, I'm ok with that.

Agreed.


>Other than that there's currently no fixing of our randomized
>upper-lower casing for function names and commands, which I would
>really like.

**Comment by renemilk** [comment]
> Assuming this is doable with our pylicense code, I'm ok with that.
It is: https://github.com/dune-community/pylicense/pull/4

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
+#   https://github.com/dune-community/dune-xt-common
+# Copyright 2009-2018 dune-xt-common developers and contributors. All rights reserved.
+# License: Dual licensed as BSD 2-Clause License (http://opensource.org/licenses/BSD-2-Clause)
+# Authors:
+#   Felix Schindler (2012 - 2017)
+#   Rene Milk       (2010 - 2018)
+#   Sven Kaulmann   (2013)
+#   Tobias Leibner  (2016)
+#
 # dune-xt-common developers and contributors. All rights reserved. License: Dual licensed as BSD 2-Clause License
```
Comment (ftalbrecht) [inline CMakeLists.txt]: Looks like a broken header from the past which got interpreted as a comment. Could you manually remove those?

File: CTestConfig.cmake [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
+# ~~~
+# This file is part of the dune-xt-common project:
+#   https://github.com/dune-community/dune-xt-common
+# Copyright 2009-2018 dune-xt-common developers and contributors. All rights reserved.
+# License: Dual licensed as BSD 2-Clause License (http://opensource.org/licenses/BSD-2-Clause)
+# Authors:
+#   Felix Schindler (2016 - 2017)
+#   Rene Milk       (2015, 2018)
+#
 # dune-xt-common developers and contributors. All rights reserved. License: Dual licensed as BSD 2-Clause License
```
Comment (ftalbrecht) [inline CTestConfig.cmake]: s.a.

File: cmake/UseDoxygen.cmake [hunk 3]
```diff
@@ -1,10 +1,19 @@
...
+# ~~~
+# This file is part of the dune-xt-common project:
+#   https://github.com/dune-community/dune-xt-common
+# Copyright 2009-2018 dune-xt-common developers and contributors. All rights reserved.
+# License: Dual licensed as BSD 2-Clause License (http://opensource.org/licenses/BSD-2-Clause)
+# Authors:
+#   Felix Schindler (2014, 2016 - 2017)
+#   Rene Milk       (2010, 2012, 2015, 2018)
+#
 # dune-xt-common developers and contributors. All rights reserved. License: Dual licensed as BSD 2-Clause License
```
Comment (ftalbrecht) [inline cmake/UseDoxygen.cmake]: s.a.

**Comment by ftalbrecht** [comment]
Forget my comments from a few moments ago. It looks like a3bc44dbe72f43b00984275b749ac3e5154cd9b7 malformed the header and 6d3c8cd2c585f7995ca244cc203be0cd8d04a17b then kept those as it interpreted them as a comment. My guess is that you should rebase on d6c7991f02762f4fcbe8220822787e7b924e028a, apply the new pylicense first, then the cmake format. Otherwise LGTM!

**Comment by renemilk** [comment]
SHould look better now, @ftalbrecht

**Comment by ftalbrecht** [comment]
Agreed! Is this ready to merge then?

**Comment by renemilk** [comment]
Kinda. I would like to add using https://github.com/cheshirekow/cmake_format/issues/29 first.
Also I need to restrict the install version to ```>= 0.3.6``` since that has the fix and new stuff.

**Comment by renemilk** [comment]
Now I'm happy, @ftalbrecht : https://github.com/dune-community/dune-xt-common/pull/67/commits/a83ba5fb188fccb25fdd246b9b8975008fab173d

**Pull request merged**